From 4fc918652dbce8c832d4f4163af87f9c6fd9ea92 Mon Sep 17 00:00:00 2001
From: "jonathan.md" <github.com.starship007@passmail.net>
Date: Sun, 1 Feb 2026 18:25:50 +0100
Subject: [PATCH] add test html

---
 src/invidious/views/template.ecr | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/invidious/views/template.ecr b/src/invidious/views/template.ecr
index 0e0f2e16..eeb36dd6 100644
--- a/src/invidious/views/template.ecr
+++ b/src/invidious/views/template.ecr
@@ -151,6 +151,9 @@
                             <a href="https://invidious.io/donate/"><%= translate(locale, "footer_donate_page") %></a>
                         </span>
                         <span>
+
+                          <div>testing</div>
+
                             <%= translate(locale, "Current version: ") %>
                             <% if CONFIG.modified_source_code_url %>
                                 <a href="<%= CONFIG.modified_source_code_url %>/commit/<%= CURRENT_COMMIT %>"><%= CURRENT_VERSION %>-<%= CURRENT_COMMIT %></a>
-- 
2.52.0

