From d79fa9b06bbdd34d8d4b53e34cf3ac63779e54ac Mon Sep 17 00:00:00 2001
From: jonathan <github.com.starship007@passmail.net>
Date: Sun, 22 Mar 2026 10:04:11 +0100
Subject: [PATCH] apply meta args

---
 docker/Dockerfile |  7 ++++---
 src/invidious.cr  | 10 +++++-----
 2 files changed, 9 insertions(+), 8 deletions(-)

diff --git a/docker/Dockerfile b/docker/Dockerfile
index e2d30364..b8caa27b 100644
--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -26,6 +26,10 @@ RUN apk add --no-cache sqlite-static yaml-static
 RUN apk del openssl-dev openssl-libs-static
 
 ARG release
+ARG CURRENT_COMMIT
+ARG CURRENT_BRANCH
+ARG CURRENT_VERSION
+ARG CURRENT_TAG
 
 WORKDIR /invidious
 COPY ./shard.yml ./shard.yml
@@ -33,9 +37,6 @@ COPY ./shard.lock ./shard.lock
 RUN shards install --production
 
 COPY ./src/ ./src/
-# TODO: .git folder is required for building – this is destructive.
-# See definition of CURRENT_BRANCH, CURRENT_COMMIT and CURRENT_VERSION.
-COPY ./.git/ ./.git/
 
 # Required for fetching player dependencies
 COPY ./scripts/ ./scripts/
diff --git a/src/invidious.cr b/src/invidious.cr
index d7c5b80b..acb44925 100644
--- a/src/invidious.cr
+++ b/src/invidious.cr
@@ -70,15 +70,15 @@ end
 HOST_URL           = make_host_url(Kemal.config)
 MAX_ITEMS_PER_PAGE = 1500
 
-CURRENT_BRANCH  = {{ "#{`git branch | sed -n '/* /s///p'`.strip}" }}
-CURRENT_COMMIT  = {{ "#{`git rev-list HEAD --max-count=1 --abbrev-commit`.strip}" }}
-CURRENT_VERSION = {{ "#{`git log -1 --format=%ci | awk '{print $1}' | sed s/-/./g`.strip}" }}
-CURRENT_TAG     = {{ "#{`git tag --points-at HEAD`.strip}" }}
+CURRENT_BRANCH  = {{ env("CURRENT_BRANCH") || "unknown" }}
+CURRENT_COMMIT  = {{ env("CURRENT_COMMIT") || "unknown" }}
+CURRENT_VERSION = {{ env("CURRENT_VERSION") || "unknown" }}
+CURRENT_TAG     = {{ env("CURRENT_TAG") || "unknown" }}
 
 # This is used to determine the `?v=` on the end of file URLs (for cache busting). We
 # only need to expire modified assets, so we can use this to find the last commit that changes
 # any assets
-ASSET_COMMIT = {{ "#{`git rev-list HEAD --max-count=1 --abbrev-commit -- assets`.strip}" }}
+ASSET_COMMIT = {{ env("CURRENT_COMMIT") || "unknown" }}
 
 SOFTWARE = {
   "name"    => "invidious",
-- 
2.53.0

